java/src/main/java/com/adhoc/flight/client/AdhocFlightClient.java

```diff
@@ -336,13 +363,27 @@ public void runQuery(final String query,
                        final @Nullable File fileToSaveTo,
                        final boolean printToConsole) throws Exception {
 
+    if (projectId != null) {
+      SetSessionOptionsResult res1 = client.setSessionOptions(
+          createSetSessionOption(PROJECT_ID_KEY, projectId), bearerToken, headerCallOption);
+      if (res1.hasErrors()) {
```

Comment:
Should we transfer this exception out from here and not write smth in the out stream?

Reply:
Converted errors into Exception message.